addons/web/static/tests/core/utils/strings_tests.js

```diff
@@ -62,4 +64,37 @@ QUnit.module("utils", () => {
         assert.deepEqual(sprintf("Hello %s!"), "Hello %s!");
         assert.deepEqual(sprintf("Hello %(value)s!"), "Hello %(value)s!");
     });
+
+    QUnit.test("sprintf properly formats numbers", (assert) => {
+        assert.deepEqual(sprintf("Hello %s!", 5), "Hello 5!");
+        assert.deepEqual(sprintf("Hello %s and %s!", 9, 10), "Hello 9 and 10!");
+        assert.deepEqual(sprintf("Hello %(x)s!", { x: 11 }), "Hello 11!");
+        assert.deepEqual(sprintf("Hello %(x)s and %(y)s!", { x: 12, y: 13 }), "Hello 12 and 13!");
+    });
+
+    QUnit.test("sprintf set behavior when value is an Array", (assert) => {
+        assert.deepEqual(sprintf("Hello %s!", ["inarray"]), "Hello inarray!");
+        assert.deepEqual(sprintf("Hello %s and %s!", [9, "10"], [11]), "Hello 9,10 and 11!");
+        assert.deepEqual(sprintf("Hello %(x)s!", { x: [11] }), "Hello 11!");
+        assert.deepEqual(
+            sprintf("Hello %(x)s and %(y)s!", { x: [12], y: ["13"] }),
+            "Hello 12 and 13!"
+        );
+    });
```

Comment:
weird to support that, but ok I guess. In the docstring we only talk about strings as values (or set of strings)